tools/mc-java/src/test/java/io/spine/tools/mc/java/gradle/McJavaExtensionDirsTest.java

```diff
@@ -0,0 +1,154 @@
+        repositories.mavenLocal();
+        repositories.mavenCentral();
+        project.getPluginManager()
+               .apply(MC_JAVA_GRADLE_PLUGIN_ID);
+    }
+
+    @Nested
+    @DisplayName("for `dirsToClean`")
+    class DirsToClean {
+
+        @Nested
+        @DisplayName("return")
+        class Return {
+            
+            @Test
+            @DisplayName("default value, if not set")
+            void defaultValue() {
+                List<String> actualDirs = actualDirs();
+
+                assertThat(actualDirs).hasSize(1);
+                assertNotEmptyAndIsInProjectDir(actualDirs.get(0));
+            }
+
+            @Test
+            @DisplayName("single value, if set")
+            void singleValue() {
+                spineProtobuf().dirToClean = newUuid();
+
+                List<String> actualDirs = actualDirs();
+
+                assertThat(actualDirs).hasSize(1);
+                assertThat(actualDirs.get(0))
+                        .isEqualTo(spineProtobuf().dirToClean);
+            }
+
+            @Test
+            @DisplayName("list, if array is set")
+            void list() {
+                spineProtobuf().dirsToClean = newArrayList(newUuid(), newUuid());
+
+                List<String> actualDirs = actualDirs();
+
+                assertThat(actualDirs)
+                        .isEqualTo(spineProtobuf().dirsToClean);
+            }
+
+            @Test
+            @DisplayName("list, if array and single are set")
+            void listIfArrayAndSingle() {
+                spineProtobuf().dirsToClean = newArrayList(newUuid(), newUuid());
+                spineProtobuf().dirToClean = newUuid();
+
+                List<String> actualDirs = actualDirs();
+
+                assertThat(actualDirs)
+                        .isEqualTo(spineProtobuf().dirsToClean);
+            }
+        }
+
+        @Test
+        @DisplayName("include `.spine` dir, if exists")
+        void includeSpineDir() throws IOException {
+            DefaultJavaPaths defaultProject = DefaultJavaPaths.at(projectDir);
+            File spineDir = defaultProject.tempArtifacts();
+            assertTrue(spineDir.mkdir());
+            String generatedDir =
+                    defaultProject.generated()
+                                  .path()
+                                  .toFile()
+                                  .getCanonicalPath();
+
+            List<String> dirsToClean = actualDirs();
+
+            assertThat(dirsToClean)
+                 .containsAtLeast(spineDir.getCanonicalPath(), generatedDir);
+        }
+
+        private List<String> actualDirs() {
+            return McJavaExtension.getDirsToClean(project);
+        }
+    }
+
+    private void assertNotEmptyAndIsInProjectDir(String path) {
+        assertFalse(path.trim()
+                        .isEmpty());
+        assertTrue(path.startsWith(project.getProjectDir()
+                                          .getAbsolutePath()));
```

Comment:
We can replace this `true` assertion with a better one from `Truth`:
```
assertThat(path)
    .startsWith(project.getProjectDir().toString())
```

In case of a failure it'll immediately provide the different and not just tell that smth is not `true`.